test/functional/adb-e2e-specs.js

```diff
@@ -0,0 +1,236 @@
+// transpile:mocha
+
+import ADB from "../../adb";
+import path from 'path';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { fs } from 'appium-support';
+import { CONNECTION_TYPES } from '../../lib/constants';
+import { sleep } from 'asyncbox';
+
+process.env.NODE_ENV = 'test';
+
+// const should = chai.should();
+// const expect = chai.expect;
+chai.use(chaiAsPromised);
+chai.should();
+
+// these test require a device connected via usb
+describe('node-adb-client', () => {
+  let device = null;
+  let availableDevices = null;
+  const packageName = "com.example.android.contactmanager";
+  const activityName = ".ContactManager";
+  before(async () => {
+    availableDevices = ADB.findAdbDevices();
+    // just select the first device
+    device = new ADB(CONNECTION_TYPES.USB, availableDevices[0]);
+    await device.connect();
+  });
+  after(async () => {
+   await device.closeConnection();
+  });
+
+  describe('shell tests', async () => {
+    it('should return a not found message for an unknown command', async () => {
+      let commandString = "asdf";
+      let expectedReturnString = `/system/bin/sh: ${commandString}: not found`;
+      let command = { // set print to false so we get the data back as a string
+        type: "shell"
+      , string: commandString
+      , print: false
+      };
+      let output = await device.runCommand(command);
+      output.indexOf(expectedReturnString).should.not.equal(-1);
+    });
+    it('should return an error message if we run a shell command incorrectly', async () => {
+      let commandString = "touch";
+      let expectedReturnString = "touch: no file specified";
+      let command = { // set print to false so we get the data back as a string
+        type: "shell"
+      , string: commandString
+      , print: false
+      };
+      let output = await device.runCommand(command);
+      output.indexOf(expectedReturnString).should.not.equal(-1);
+    });
+    it('should return successful output if we run a shell command correctly', async () => {
+      let commandString = "cd sdcard; pwd";
+      let expectedReturnString = "/sdcard";
+      let command = { // set print to false so we get the data back as a string
+        type: "shell"
+      , string: commandString
+      , print: false
+      };
+      let output = await device.runCommand(command);
+      output.indexOf(expectedReturnString).should.not.equal(-1);
+    });
+  });
+  describe('push tests', () => {
+    const smallFile = path.resolve(__dirname, '..', '..', '..', 'test', 'fixtures', 'smallFile');
+    const largeFile = path.resolve(__dirname, '..', '..', '..', 'test', 'fixtures', 'largeFile');
```

Comment:
It's current 191k, it needs to be at least larger than 64k in order to test the protocols handling files properly. I'll look at generating a temp. file instead of cloning this file all over the place.